limiter_atomic_int64.go

```diff
@@ -66,10 +66,10 @@ func (t *atomicInt64Limiter) Take() time.Time {
 		timeOfNextPermissionIssue := atomic.LoadInt64(&t.state)
 
 		switch {
-		case timeOfNextPermissionIssue == 0:
-			// If this is our first request, then we allow it.
+		case t.maxSlack == 0 && now-timeOfNextPermissionIssue > int64(t.perRequest):
```

Comment:
I still need to grok the code, but briefly looking at this, I wonder if we should do:

```golang
if t.maxSlack == 0 {
  t.maxSlack == t.preRequest
}
```
during the initialization to get rid of one of the cases.

Reply:
Unfortunately, no, because this will allow for one additional permission to accumulate.
In general, I think we can simplify this and other implementations if we change the way how we initialize rate limiter state, but this is a subject for a separate PR, I think.

Reply:
Hm. I will need to really think about this more.

In your current proposal though, are we not changing the behaviour for the very first request?

I might add tests for the first request only to formalize the behavior.

Reply:
I'm less worried about implementation simplification than any unspecified behavior - I'd like to make sure that all implementation behave identically.

Tried to address initial few "Takes" in #97 - WDYT?

I might add one or two more tests before looking at the "time" changes.

Reply:
> I'm less worried about implementation simplification than any unspecified behavior - I'd like to make sure that all implementation behave identically.

This fix functional behaves identically to the previous behavior. It just applies the same behavior to cases when no one has used RL for a long time, so it accumulated many permissions. But this limiter has no slack, so we should not allow permissions to accumulate, so we need to move the `newTimeOfNextPermissionIssue` to `now`, as if RL was created right now.